units/migrate_volume.go

```diff
@@ -99,7 +99,7 @@ func (j *volumeMigrationJob) Run(ctx context.Context) {
 		return
 	}
 
-	if j.initialHost.HomeVolumeID != "" {
+	if j.volume.Host != "" {
```

Comment:
This way seems less idempotent than before if we're assuming the job may re-run because L103 will unset the volume's host, but then could fail to unset the host's home volume on L109, resulting in a broken one-way link between the initial host and the volume. Is it a problem if this happens?

Reply:
It's true, but this pattern is with precent in the evergreen codebase; if a workstation is terminated, it still points to the volume, but the volume is no longer associated with the host. It's still not ideal though.

One other approach I considered was setting a flag if we find initially that the volume is unattached to a host (i.e. the host was terminated). Then we could more easily skip the "start new host" and "detach old host" stages since they're already done. Would be interested in any thoughts you have on that! 🙏

Reply:
I think this change looks good. It should be okay to re-run the start new host/detach old host checks since they're idempotent anyways, so even if there is a bit of unnecessary repeated work, it's not a big deal.